src/com/trilead/ssh2/crypto/dh/DhExchange.java

```diff
@@ -1,147 +1,129 @@
-
 package com.trilead.ssh2.crypto.dh;
 
-import java.io.UnsupportedEncodingException;
+import java.io.IOException;
 import java.math.BigInteger;
-import java.security.SecureRandom;
-
-import com.trilead.ssh2.crypto.digest.HashForSSH2Types;
-import com.trilead.ssh2.log.Logger;
-
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+
+import javax.crypto.KeyAgreement;
+import javax.crypto.interfaces.DHPrivateKey;
+import javax.crypto.interfaces.DHPublicKey;
+import javax.crypto.spec.DHParameterSpec;
+import javax.crypto.spec.DHPublicKeySpec;
 
 /**
- * DhExchange.
- * 
- * @author Christian Plattner, plattner@trilead.com
- * @version $Id: DhExchange.java,v 1.2 2008/04/01 12:38:09 cplattne Exp $
+ * @author kenny
+ *
  */
-public class DhExchange
-{
-	private static final Logger log = Logger.getLogger(DhExchange.class);
+public class DhExchange extends GenericDhExchange {
 
 	/* Given by the standard */
 
-	static final BigInteger p1, p14;
```

Comment:
Removing these fields introduces binary incompatibility. It looks like they've been replaced with `P1` and `P14`, so can we not keep the old fields, deprecate them, and make them equal to the value of the new fields?